sale_channel/models/sale_order.py

```diff
@@ -0,0 +1,15 @@
+#  Copyright (c) Akretion 2020
+#  License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html)
+
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    sale_channel_id = fields.Many2one("sale.channel", ondelete="set null")
+
+    def _finalize_invoices(self, invoices, references):
```

Comment:
do not inherit this method.
The invoice is already created here so we are going to do an extra write operation on the object invoice.

It's better to inherit the method "_prepare_invoice" in order to inject the sale channel and avoid useless additionnal write. See prepare method here : https://github.com/OCA/OCB/blob/12.0/addons/sale/models/sale.py#L460